### ISSUE TITLE
Unprocessable Entity during image upload

### ISSUE THREAD

**Issue by chenny408** [volmer/bootsy#108]
Started POST "/bootsy/image_galleries/1/images" for 127.0.0.1 at 2014-12-16 12:28:52 -0800
Processing by Bootsy::ImagesController#create as JSON
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"wPYDPnQlLTDQpJ91tYyZfusfY0aIr4BVbwFwTxMuOVo=", "remotipart_submitted"=>"true", "X-Requested-With"=>"IFrame", "X-Http-Accept"=>"application/json, text/javascript, */*; q=0.01", "image"=>{"image_file"=>#<ActionDispatch::Http::UploadedFile:0x007fd121418648 @tempfile=#<Tempfile:/var/folders/jn/zx3nhkqd4gd38grv_04jslgw0000gn/T/RackMultipart20141216-50530-8nldl6>, @original_filename="team3.jpg", @content_type="image/jpeg", @headers="Content-Disposition: form-data; name=\"image[image_file]\"; filename=\"team3.jpg\"\r\nContent-Type: image/jpeg\r\n">}, "image_gallery_id"=>"1"}
  Bootsy::ImageGallery Load (0.6ms)  SELECT  "bootsy_image_galleries".* FROM "bootsy_image_galleries"  WHERE "bootsy_image_galleries"."id" = $1 LIMIT 1  [["id", 1]]
   (0.7ms)  BEGIN
   (0.4ms)  ROLLBACK
Completed 422 Unprocessable Entity in 305ms (Views: 0.5ms | ActiveRecord: 2.9ms)

It seems like the image cannot be saved? 

one issue i ran into during the installation was that 'rake bootsy:install:migrations' didnt return anything, so i had to manually add the tables into the migration. not sure if this affects the database write.

**Comment by dhampik** [comment]
Any update on this? Is it true that image upload is broken in latest version?

**Comment by volmer** [comment]
I couldn't reproduce this problem with 2.0.13 or master. The image upload works fine for me. Could anyone please give me more details on this? More accurate reproduction steps or maybe a sample application?

**Comment by chenny408** [comment]
no, turns out there was an issue with my environment, i believe with the ImageMagick plugin that converts the formatting.

**Comment by driehuis** [comment]
I was able to reproduce this issue with the sample app from issue #105 (for which I applied the suggested fix, and updated the gems; my copy is here: https://github.com/driehuis/bootsy-sample).

I tried uploading a .BMP, which was accepted by the JavaScript and uploaded. From the user perspective, the page then just sat there, with a throbber suggesting that the server was processing.

I see two issues here:

* The exception was not passed down to JavaScript, so the user has no idea it bombed with this:
```
Started POST "/bootsy/image_galleries/2/images" for 127.0.0.1 at 2015-03-04 15:38:41 +0100
Processing by Bootsy::ImagesController#create as JSON
  Parameters: {"utf8"=>"✓", "authenticity_token"=>"nVXLNFSOmwe/r1TPEYyJykZws5paoFDV4mxJ6p4JjfI=", "remotipart_submitted"=>"true", "X-Requested-With"=>"IFrame", "X-Http-Accept"=>"application/json, text/javascript, */*; q=0.01", "image"=>{"image_file"=>#<ActionDispatch::Http::UploadedFile:0x007f9b14cbe218 @tempfile=#<Tempfile:/tmp/RackMultipart20150304-18191-198hnko>, @original_filename="test-320x240.bmp", @content_type="image/bmp", @headers="Content-Disposition: form-data; name=\"image[image_file]\"; filename=\"test-320x240.bmp\"\r\nContent-Type: image/bmp\r\n">}, "image_gallery_id"=>"2"}
  Bootsy::ImageGallery Load (0.1ms)  SELECT  "bootsy_image_galleries".* FROM "bootsy_image_galleries"  WHERE "bootsy_image_galleries"."id" = ? LIMIT 1  [["id", 2]]
   (0.1ms)  begin transaction
   (0.1ms)  commit transaction
   (0.1ms)  begin transaction
   (0.1ms)  rollback transaction
Completed 422 Unprocessable Entity in 7ms (Views: 0.3ms | ActiveRecord: 0.5ms)
```

* When creating a new Post, with a .jpg and a .bmp uploaded and bombing on the second, an orphaned ImageGallery was created, not referring back to the original Post. Not that this orphan hurts a lot, but from the user perspective, after hitting Refresh, the image tray was empty and the successfully uploaded .jpg inaccessible.
```
=> #<Bootsy::ImageGallery id: 1, bootsy_resource_id: nil, bootsy_resource_type: "Post", created_at: "2015-03-04 14:08:38", updated_at: "2015-03-04 14:09:01">
```

**Comment by volmer** [comment]
Hey @driehuis, thanks for the detailed report. I could in fact reproduce these issues in `2.0.13`, but it looks like it's fixed in the master branch. Could you please verify? I'm planning to release a new version of the gem soon.

**Comment by driehuis** [comment]
Confirmed fixed in 2.1.0. Thanks a bundle!

**Comment by shybovycha** [comment]
Using 2.4.0 and still able to reproduce the issue
